### ISSUE TITLE
Print both training and validation loss

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/litgpt#1042]
Right now we print the output as

```
iter 1760 | step 220: loss 1.5947, iter time: 263.85 ms (optimizer.step)
iter 1770 | step 221: loss 1.5191, iter time: 89.96 ms
iter 1780 | step 222: loss 1.5989, iter time: 89.64 ms
iter 1790 | step 223: loss 1.4138, iter time: 86.21 ms
iter 1800 | step 225: loss 1.8572, iter time: 257.35 ms (optimizer.step)
iter 1810 | step 226: loss 1.6535, iter time: 97.41 ms
iter 1820 | step 227: loss 2.1728, iter time: 104.73 ms
```

A new user would expect this to be the training loss because it's the simpler one. But I think if we explicitely make it a `train loss` and `val loss`, it would be easier to know what it is. 

Also, having both train and val loss helps with detecting issues during finetuning. 
E.g., here I would have to scroll up to see the val loss:

```
iter 4800: val loss 1.7258, val time: 3778.02 ms
iter 4810 | step 601: loss 0.2745, iter time: 131.68 ms
iter 4820 | step 602: loss 0.2813, iter time: 60.50 ms
iter 4830 | step 603: loss 0.2746, iter time: 69.50 ms
iter 4840 | step 605: loss 0.2609, iter time: 153.14 ms (optimizer.step)
iter 4850 | step 606: loss 0.3222, iter time: 61.86 ms
```

If it is more like this:

```
iter 4800: val loss 1.7258, val time: 3778.02 ms
iter 4810 | step 601: loss train 0.274, val 1.725 | iter time: 131.68 ms
iter 4810 | step 601: loss train 0.260, val 1.725 | iter time: 131.68 ms
iter 4810 | step 601: loss train 0.800, val 1.725 | iter time: 131.68 ms
...
```

It might be more readble

If you see the train loss go down while the val loss goes up, you know you are overfitting.

If you see both train loss and val loss go up, you know it's likely due to a bad learning rate, batch size, etc.
